shiny/ui/_page.py

```diff
@@ -443,6 +444,99 @@ def page_bootstrap(
     )
 
 
+def page_auto(
+    *args: TagChild | TagAttrs,
+    _page_fn: Callable[..., Tag] | None = None,
+    _fillable: bool = False,
+    _full_width: bool = False,
+    **kwargs: object,
+) -> Tag:
+    """
+    A page container which automatically decides which page function to use.
+
+    If there is a top-level nav, this will use :func:`~shiny.ui.page_navbar`. If not,
+    and there is a top-level sidebar, this will use :func:`~shiny.ui.page_sidebar`.
+
+    If there are neither top-level navs nor sidebars, this will use the ``_fillable``
+    and ``_full_width`` arguments to determine which page function to use.
+
+    Parameters
+    ----------
+    *args
+        UI elements. These are used to determine which page function to use, and they
+        are also passed along to that page function.
+    _page_fn
+        The page function to use. If ``None`` (the default), will automatically choose
+        one based on the arguments provided.
+    _fillable
+        This has an effect only if there are no sidebars or top-level navs. If ``True``,
```

Comment:
`page_navbar` and `page_sidebar` both take a `fillable` argument, we could pass it through